### PR TITLE
add support to set UiAutomator Congfigurator values

### PR DESCRIPTION
**Pull request by truebit**
## Proposed changes

[UiAutomator Configurator](https://developer.android.com/reference/android/support/test/uiautomator/Configurator.html) provides several methods to change its the configurations.
I have implemented them in appium using [appium settings api](https://github.com/appium/appium/tree/master/docs/en/advanced-concepts/settings.md)

This is the overall PR for all other three PRs:
[appium-android-driver](https://github.com/appium/appium-android-driver/pull/153)
[appium-android-bootstrap](https://github.com/appium/appium-android-bootstrap/pull/25)
[java-client](https://github.com/appium/java-client/pull/410)

I know there would be `appium-uiautomator2-server` and `appium-uiautomator2-driver` in upcoming versions. Since I used existing API, It would be OK for me to merge this feature to the new android server and driver
## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://docs.google.com/forms/d/1lOfXRw_0VCk7gYzjj4WLetGu7yelDVo5LWh0z7pGftE/viewform)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules
## Further comments

But according to [android uiautomator fixed timeout bug in UiAutomatorBridge](https://code.google.com/p/android/issues/detail?id=73297), the timeout in dynamic page still cannot be changed.
Maybe we need to recompile `uiautomator.jar` to change the static final field [`TOTAL_TIME_TO_WAIT_FOR_IDLE_STATE`](https://android.googlesource.com/platform/frameworks/uiautomator/+/master/src/com/android/uiautomator/core/UiAutomatorBridge.java#39) from `10 * 1000` to `Configurator.getInstance().getWaitForIdleTimeout()`.
That's a little more complexed, I have not tried that.

### CODE REVIEW

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by triager** [comment]
Can one of the admins verify this patch?

File: docs/en/advanced-concepts/settings.md [hunk 1]
```diff
@@ -38,3 +38,5 @@ Settings are implemented via the following API endpoints:
 ### Supported Settings
 
 **"ignoreUnimportantViews"** - Boolean which sets whether Android devices should use `setCompressedLayoutHeirarchy()` which ignores all views which are marked IMPORTANT_FOR_ACCESSIBILITY_NO or IMPORTANT_FOR_ACCESSIBILITY_AUTO (and have been deemed not important by the system), in an attempt to make things less confusing or faster.
+
+**"configurator"** - JSON string (example: `{"method":"setWaitForIdleTimeout","value":5000}`) which sets [UiAutomator Configurator](https://developer.android.com/reference/android/support/test/uiautomator/Configurator.html) timeouts and delays in Android devices.
```
Comment (jlipps) [inline docs/en/advanced-concepts/settings.md]: any reason it couldn't just be a JSON object that gets sent rather than a JSON string?

Comment (truebit) [inline docs/en/advanced-concepts/settings.md]: I think in http request, all params are treated as String. In client implementation, we could parse to JSON Object

**Comment by jlipps** [comment]
👍 just squash please!

**Comment by truebit** [comment]
It seems that because I merged from upstream master, the squash included some changes in other upstream commits. Is that OK?

**Comment by jlipps** [comment]
@truebit, no, this PR should contain only your changes! looks like you need to rebase on upstream master instead of merging from it

**Comment by truebit** [comment]
sqash done as you said :)

**Comment by truebit** [comment]
@imurchie You merged this PR, but left https://github.com/appium/appium-android-driver/pull/153 unaccepted. without the left PR, it won't work.

**Comment by imurchie** [comment]
Yes. Once everyone is happy with that PR, it will be merged. This one is just documentation, and being slightly out of sync is not too much of a problem. Especially since the work will only be available to someone using the source code anyway, until we have time to release a new version.

**Pull request merged**